core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -993,6 +1000,15 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleApiVersionRequest(request: RequestChannel.Request) {
+    val responseHeader = new ResponseHeader(request.header.correlationId)
+    val responseBody = if (!authorize(request.session, Describe, Resource.ClusterResource))
+      ApiVersionResponse.fromError(Errors.CLUSTER_AUTHORIZATION_FAILED)
```

Comment:
I think the security gain from restricting this API is a little dubious, but I can understand the desire to limit information leakage. Anyway, perhaps there's no need to rehash the argument since the KIP already says this?

> The broker returns its full list of supported ApiKeys and versions regardless of current authentication state (e.g., before SASL authentication). If this is considered to leak information SSL can be used for early authentication.